morpheus/models/dfencoder/autoencoder.py

```diff
@@ -1010,112 +868,94 @@ def _fit_distributed(
-            train_df = train_data  # keep the dataframe for model-building
-            train_data = DatasetFromDataframe(
-                df=train_data,
-                batch_size=self.batch_size,
-                preprocess_fn=self.preprocess_train_data,
-                shuffle_rows_in_batch=True,
-            )
+        if (self.optim is None):
+            self._build_model(df=train_data, rank=rank)
 
-        if self.optim is None:
-            self._build_model(df=train_df, rank=rank)
+        train_dset, val_dset, loss_dset = self._build_training_datasets(train_data, val_data, run_validation,
+                                                                        use_val_for_loss_stats)
 
-        is_main_process = rank == 0
-        should_run_validation = (run_validation and val_data is not None)
-        if self.patience and not should_run_validation:
+        is_main_process = (rank == 0)
+
+        # Early stopping warning
+        if (self.patience and not run_validation):
             LOG.warning(
                 f"Not going to perform early-stopping. self.patience(={self.patience}) is provided for early-stopping"
                 " but validation is not enabled. Please set `run_validation` to True and provide a `val_dataset` to"
                 " enable early-stopping.")
 
-        if is_main_process and should_run_validation:
+        # Check if we're running validation, and compute baseline performance if we are
+        if (is_main_process and run_validation):
             LOG.debug('Validating during training. Computing baseline performance...')
-            baseline = self._compute_baseline_performance_from_dataset(val_data)
+            baseline = self._compute_baseline_performance_from_dataset(val_dset)
 
             if isinstance(self.logger, BasicLogger):
                 self.logger.baseline_loss = baseline
 
             LOG.debug(f'Baseline loss: {round(baseline, 4)}')
 
-        # early stopping
-        count_es = 0
-        last_val_loss = float('inf')
-        should_early_stop = False
-        for epoch in range(epochs):
-            LOG.debug(f'Rank{rank} training epoch {epoch + 1}...')
-
-            # if we are using DistributedSampler, we have to tell it which epoch this is
-            train_data.sampler.set_epoch(epoch)
-
-            train_loss_sum = 0
-            train_loss_count = 0
-            for data_d in train_data:
-                loss = self._fit_batch(**data_d['data'])
+        # Training loop
+        self._train_for_epochs(train_dset, val_dset, epochs, rank, world_size, run_validation, is_main_process)
 
-                train_loss_count += 1
-                train_loss_sum += loss
+        if (is_main_process):
+            loss_dset.convert_to_validation(self)
+            self._populate_loss_stats_from_dataset(loss_dset)
 
-            if self.lr_decay is not None:
-                self.lr_decay.step()
+    def _validate_dataframe(self, orig_df, swapped_df):
```

Comment:
We don't seem to be using this function anymore, should we remove it altogether?